### PR TITLE
kubeadm: increase ut coverage for util/version

### PR DESCRIPTION
**Pull request by my-git9**
#### What type of PR is this?

kubeadm: increase ut coverage for util/version

before:

image

after the change:

image

/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NONE
```

### CODE REVIEW

File: cmd/kubeadm/app/util/version_test.go [hunk 1]
```diff
@@ -525,3 +528,77 @@ func TestValidateStableVersion(t *testing.T) {
...
+			result, err := fetchFromURL(url, tt.timeout)
+
+			if tt.expected != "" {
+				if err != nil {
+					t.Errorf("Expected no error but got: %v", err)
+				}
+				if result != tt.expected {
+					t.Errorf("Expected '%s', but got '%s'", tt.expected, result)
+				}
+			} else if tt.expected != "" && err == nil {
```
Comment (googs1025) [inline cmd/kubeadm/app/util/version_test.go:599]: If `tt.expected != ""` the else if condition will never be executed

Comment (my-git9) [reply cmd/kubeadm/app/util/version_test.go]: Adjusted, thanks

File: cmd/kubeadm/app/util/version_test.go [hunk 2]
```diff
@@ -525,3 +528,77 @@ func TestValidateStableVersion(t *testing.T) {
...
+			result, err := fetchFromURL(url, tt.timeout)
+
+			if tt.expected != "" {
+				if err != nil {
+					t.Errorf("Expected no error but got: %v", err)
+				}
+				if result != tt.expected {
+					t.Errorf("Expected '%s', but got '%s'", tt.expected, result)
+				}
+			} else if tt.expected == "" && err == nil {
```
Comment (carlory) [inline cmd/kubeadm/app/util/version_test.go:599]: make it simpler

```suggestion
			} else if err == nil {
```

File: cmd/kubeadm/app/util/version_test.go [hunk 3]
```diff
@@ -525,3 +528,77 @@ func TestValidateStableVersion(t *testing.T) {
...
+						t.Error("Write body failed.")
+					}
+				}
+			})
+
+			ts := httptest.NewServer(handler)
+			defer ts.Close()
+
+			url := ts.URL + tt.url
+
```
Comment (carlory) [inline cmd/kubeadm/app/util/version_test.go:589]: remove blank line

File: cmd/kubeadm/app/util/version_test.go [hunk 4]
```diff
@@ -525,3 +528,77 @@ func TestValidateStableVersion(t *testing.T) {
...
+			result, err := fetchFromURL(url, tt.timeout)
+
+			if tt.expected != "" {
+				if err != nil {
+					t.Errorf("Expected no error but got: %v", err)
+				}
+				if result != tt.expected {
+					t.Errorf("Expected '%s', but got '%s'", tt.expected, result)
```
Comment (carlory) [inline cmd/kubeadm/app/util/version_test.go:597]:
```suggestion
					t.Errorf("Expected %q, but got %q", tt.expected, result)
```

File: cmd/kubeadm/app/util/version_test.go (hunk 2 above)
Comment (my-git9) [reply cmd/kubeadm/app/util/version_test.go]: Adjusted, thanks

File: cmd/kubeadm/app/util/version_test.go [hunk 5]
```diff
@@ -525,3 +527,73 @@ func TestValidateStableVersion(t *testing.T) {
...
+			if tt.expected != "" {
+				if err != nil {
+					t.Errorf("Expected no error but got: %v", err)
+				}
+				if result != tt.expected {
+					t.Errorf("Expected '%q', but got '%q'", tt.expected, result)
+				}
+			} else if err == nil {
+				t.Errorf("Expected an error but got none")
+			}
```
Comment (neolit123) [inline cmd/kubeadm/app/util/version_test.go:596]: isn't it sufficient to just do the following?

```
if result != tt.expected {
	t.Errorf("Expected '%q', but got '%q'", tt.expected, result)
}
```

Comment (my-git9) [reply cmd/kubeadm/app/util/version_test.go]: Modfied, thanks

File: cmd/kubeadm/app/util/version_test.go [hunk 6]
```diff
@@ -525,3 +527,72 @@ func TestValidateStableVersion(t *testing.T) {
...
+				}
+			})
+
+			ts := httptest.NewServer(handler)
+			defer ts.Close()
+
+			url := ts.URL + tt.url
+			result, err := fetchFromURL(url, tt.timeout)
+			if (err != nil) != tt.expectErr {
+				t.Errorf("ExpectErr '%v', but got '%v'", tt.expectErr, err)
```
Comment (neolit123) [inline cmd/kubeadm/app/util/version_test.go:591]:
```suggestion
				t.Errorf("expected error: %v, got: %v", tt.expectErr, err)
```

File: cmd/kubeadm/app/util/version_test.go [hunk 7]
```diff
@@ -525,3 +527,72 @@ func TestValidateStableVersion(t *testing.T) {
...
+			}
+			if tt.expected != result {
+				t.Errorf("Expected '%q', but got '%q'", tt.expected, result)
```
Comment (neolit123) [inline cmd/kubeadm/app/util/version_test.go:594]:
```suggestion
				t.Errorf("expected result: %q, got: %q", tt.expected, result)
```

**Review by neolit123 — COMMENTED** [review]
/approve
just a couple of more minor comments.

**Pull request merged**